lib/shopify_api/resources/o_auth.rb

```diff
@@ -1,3 +1,8 @@
+# frozen_string_literal: true
+
+# This resource is deprecated and will likely be removed in a future version of this gem.
+# Use ShopifyAPI::ApiPermission.destroy instead
```

Comment:
You probably want to log this when someone uses the deprecated method.

```ruby
def self.revoke
  warn "[DEPRICATION] ShopifyAPI::OAuth is deprecated and will likely be removed in a future version of this gem. Use ShopifyAPI::ApiPermission.destroy instead"
  
  delete(:revoke)
end
```
maybe?